samples/snippets/src/main/java/com/example/spanner/admin/generated/CreateInstanceConfigSample.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.spanner.admin.generated;
+
+// [START spanner_create_instance_config]
+import com.google.cloud.spanner.admin.instance.v1.InstanceAdminClient;
+import com.google.spanner.admin.instance.v1.CreateInstanceConfigRequest;
+import com.google.spanner.admin.instance.v1.InstanceConfig;
+import com.google.spanner.admin.instance.v1.ReplicaInfo;
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+class CreateInstanceConfigSample {
+  static void createInstanceConfig() throws IOException {
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "projects/my-project";
+    String baseInstanceConfig = "nam11";
+    String instanceConfigId = "custom-instance-config4";
+    String instanceConfigName = "projects/my-project/instanceConfigs/custom-instance-config4";
```

Comment:
nit: can we construct this string from the other variables in this method, so something like `String.format("projects/%s/instanceConfigs/%s",  projectId, instanceConfigId)`

Reply:
Yes, I have now modified all new samples to use the same convention (projectId, configId, instanceId) and using utilities like `ProjectName.of()`, `InstanceConfigName.of()`, `InstanceName.of()` . I see that this will be helpful as customers don't need to understand the difference between id and name.

Reply:
Thanks 👍